src/plugins/chart_expressions/expression_metric/public/components/metric_vis.tsx

```diff
@@ -307,6 +307,19 @@ export const MetricVis = ({
     pixelWidth = grid[0]?.length * maxTileSideLength;
   }
 
+  const [scrollChildHeight, setScrollChildHeight] = useState<string>('100%');
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    const minTileHeight = 64; // TODO - magic number from the @elastic/charts side. would be nice to deduplicate
+    const minimumRequiredVerticalSpace = minTileHeight * grid.length;
+    setScrollChildHeight(
+      (scrollContainerRef.current?.clientHeight ?? -Infinity) > minimumRequiredVerticalSpace
```

Comment:
As this is reading layout information which can change (browser resizing, moving panels around on the dashboard), we should use the eui resize observer to re-calculate on changes: https://elastic.github.io/eui/#/utilities/resize-observer